test/monitoring/components_alerts_test.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2020 The Lokomotive Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build aws packet
+// +build poste2e
+
+package monitoring
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"testing"
+	"time"
+
+	v1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	testutil "github.com/kinvolk/lokomotive/test/components/util"
+)
+
+const (
+	retryInterval  = time.Second * 5
+	timeout        = time.Minute * 9
+	contextTimeout = 10
+)
+
+//nolint:funlen
+func testComponentAlerts(t *testing.T, v1api v1.API) {
+	alertTestCases := []struct {
+		ComponentName string
+		RuleGroup     string
+		platforms     []testutil.Platform
+		Alerts        []string
+	}{
+		{
+			ComponentName: "metallb",
+			RuleGroup:     "metallb-rules",
+			platforms:     []testutil.Platform{testutil.PlatformPacket},
+			Alerts: []string{
+				"MetalLBNoBGPSession", "MetalLBConfigStale", "MetalLBControllerPodsAvailability",
+				"MetalLBSpeakerPodsAvailability",
+			},
+		},
+	}
+
+	for _, tc := range alertTestCases {
+		tc := tc
+		t.Run(tc.ComponentName, func(t *testing.T) {
+			t.Parallel()
+
+			if !testutil.IsPlatformSupported(t, tc.platforms) {
+				t.Skip()
+			}
+
+			retryFunc := func() (done bool, err error) {
```

Comment:
Could we move this function from inline to separate, to reduce the indentation deepness. I think it will make things more readable, as right now it's not immediately obvious to me what this function is doing.

Reply:
done